File: en/java-spring-demo/src/main/java/com/itangcent/springboot/demo/controller/FileCtrl.java
package com.itangcent.springboot.demo.controller;

import com.itangcent.springboot.demo.common.dto.IResult;
import com.itangcent.springboot.demo.common.dto.Result;
import com.itangcent.springboot.demo.common.dto.UserDto;
import com.itangcent.springboot.demo.common.model.UserInfo;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.multipart.MultipartFile;

/**
 * File Upload Controller
 * Handles user-related file upload operations, including single and multiple file uploads.
 * Supports both combined and separate submission of user information and files.
 *
 * @module user
 */
@RestController
@RequestMapping(value = "file")
public class FileCtrl {
    /**
     * Create new user and upload avatar
     * Uses @ModelAttribute to receive form data and file, implementing one-time submission of user information and avatar.
     * 
     * @param userDto Data transfer object containing user information and avatar file - UserDto contains basic user info and MultipartFile type avatar
     * @return Returns save result, returns "Save successful" message on success
     */
    @PostMapping("/add")
    public IResult add(@ModelAttribute UserDto userDto) {
        return Result.success("Save successful");
    }

    /**
     * Create new user and upload avatar (Separate parameter method)
     * User information and file are received separately, providing more flexible submission options
     *
     * @param userInfo Basic user information - Object containing username, age, and other basic information
     * @param profileImg User avatar file - Single avatar file, received using MultipartFile
     * @return Returns save result, returns "Save successful" message on success
     * @undone Interface to be improved - Need to add file size validation and type validation
     */
    @PostMapping("/add2")
    public IResult add(UserInfo userInfo,
                       @RequestParam(name = "profileImg") MultipartFile profileImg) {
        return Result.success("Save successful");
    }

    /**
     * Create new user and upload multiple avatars
     * Supports batch upload of avatar files, can upload multiple avatar images simultaneously
     *
     * @param userInfo Basic user information - Object containing username, age, and other basic information
     * @param profileImgs Multiple avatar files array - Supports uploading multiple avatar files simultaneously
     * @return Returns save result, returns "Save successful" message on success
     * @undone Interface to be improved - Need to add file quantity limit and total size limit
     */
    @PostMapping("/add3")
    public IResult add(UserInfo userInfo,
                       @RequestParam(name = "profileImg") MultipartFile[] profileImgs) {
        return Result.success("Save successful");
    }
}

File: cn/kotlin-spring-demo/src/main/kotlin/com/itangcent/springboot/kotlin/demo/controller/FileCtrl.kt
package com.itangcent.springboot.kotlin.demo.controller

import com.itangcent.common.dto.IResult
import com.itangcent.common.dto.Result
import com.itangcent.common.dto.UserDto
import com.itangcent.common.model.UserInfo
import org.springframework.web.bind.annotation.*
import org.springframework.web.multipart.MultipartFile

/**
 * 文件上传控制器
 * 提供用户相关的文件上传功能，如头像上传等
 *
 * @module user
 */
@RestController
@RequestMapping(value = ["/file"])
class FileCtrl {
    /**
     * 新增用户（包含头像上传）
     * 使用ModelAttribute接收用户信息和头像文件
     *
     * @param userDto 包含用户信息和头像文件的数据传输对象
     * @return 返回保存结果信息
     */
    @PostMapping("/add")
    fun add(@ModelAttribute userDto: UserDto): IResult<String> {
        return Result.success("保存成功")
    }

    /**
     * 新增用户（头像单独上传）
     * 演示如何分别接收用户信息和头像文件
     *
     * @param userInfo 用户基本信息
     * @param profileImg 用户头像文件
     * @return 返回保存结果信息
     * @undone 待实现
     */
    @PostMapping("/add2")
    fun add(userInfo: UserInfo,
            @RequestParam(name = "profileImg") profileImg: MultipartFile): IResult<String> {
        return Result.success("保存成功")
    }

    /**
     * 新增用户（支持多个头像）
     * 演示如何处理多文件上传的场景
     *
     * @param userInfo 用户基本信息
     * @param profileImgs 多个头像文件数组
     * @return 返回保存结果信息
     * @undone 待实现
     */
    @PostMapping("/add3")
    fun add(userInfo: UserInfo,
            @RequestParam(name = "profileImg") profileImgs: Array<MultipartFile>): IResult<String> {
        return Result.success("保存成功")
    }
}